### PR TITLE
iOS-2162 Set editor | read only

### PR DESCRIPTION
**Pull request by mgolovko**
https://github.com/anyproto/anytype-swift/assets/2483784/ed599569-159b-4692-85e2-b9fde86c5227

### CODE REVIEW

File: Anytype/Sources/PresentationLayer/Common/SwiftUI/BasicComponents/ButtonForeground.swift [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+        content
+            .foregroundColor(isEnable ? .Button.active : .Button.inactive)
+    }
+}
+
+extension View {
+    func buttonDynamicForegroundColor() -> some View {
+        modifier(ButtonForegroundViewModifier())
+    }
+}
```
Comment (ignatovv) [inline Anytype/Sources/PresentationLayer/Common/SwiftUI/BasicComponents/ButtonForeground.swift:18]: Naming not really transparent on what to expect from `buttonDynamicForegroundColor`
Also `isEnabe` and `foregroundColor` can be applied not only to buttons but other components as well
I have two optional suggestions
rename `buttonDynamicForegroundColor` to `foregroundColorForDisabledState`
or make it 
dynamicForegroundColor(enabled: Color, disabled: Color)

**Review by ignatovv — APPROVED** [review]

File: Anytype/Sources/PresentationLayer/TextEditor/Set/Views/Popups/ViewPicker/SetViewPicker.swift [hunk 2]
```diff
@@ -53,7 +56,8 @@ struct SetViewPicker: View {
         .toolbar {
             ToolbarItem(placement: .navigationBarLeading) {
                 EditButton()
-                    .foregroundColor(Color.Button.active)
+                    .buttonDynamicForegroundColor()
```
Comment (joe-pusya) [inline Anytype/Sources/PresentationLayer/TextEditor/Set/Views/Popups/ViewPicker/SetViewPicker.swift:59]: In relations we just hide buttons in toolbar in non-editable mode, maybe we should make it here in the same way?

**Review by joe-pusya — APPROVED** [review]

File: Anytype/Sources/PresentationLayer/Common/SwiftUI/BasicComponents/ButtonForeground.swift (hunk 1 above)
Comment (mgolovko) [reply Anytype/Sources/PresentationLayer/Common/SwiftUI/BasicComponents/ButtonForeground.swift:18]: discussed

File: Anytype/Sources/PresentationLayer/TextEditor/Set/Views/Popups/ViewPicker/SetViewPicker.swift (hunk 2 above)
Comment (mgolovko) [reply Anytype/Sources/PresentationLayer/TextEditor/Set/Views/Popups/ViewPicker/SetViewPicker.swift]: fixed

**Pull request merged**